test/test_logging_macros.cpp

```diff
@@ -143,9 +143,9 @@ TEST_F(TestLoggingMacros, test_logging_throttle) {
 TEST_F(TestLoggingMacros, test_logging_skipfirst_throttle) {
   for (int i : {0, 1, 2, 3, 4, 5, 6, 7, 8, 9}) {
     RCUTILS_LOG_FATAL_SKIPFIRST_THROTTLE(
-      RCUTILS_STEADY_TIME, 30 /* ms */, "throttled message %d", i)
+      RCUTILS_STEADY_TIME, 50 /* ms */, "throttled message %d", i)
     using namespace std::chrono_literals;
-    std::this_thread::sleep_for(20ms);
+    std::this_thread::sleep_for(30ms);
```

Comment:
Another option is to put the whole test in a loop, retrying a limited number of times. Presumably if it passes any of the attempts then there is no issue. This test is really testing the logic, not the timeliness, of the `sleep_for` function.

Yet another option is to measure the time you actually sleep for (measure around the `sleep_for` function call) and accumulate the total since the first call. Then each time you update the total you can figure out if the next `i` should be in the output or not. This would at least be slightly more robust to a overrun on `sleep_for` (remember it only guarantees that it returns after the period, not how close to it).